### PR TITLE
Add miscellaneous Unix artefacts

### PR DESCRIPTION
**Pull request by Poeloe**
(DIS-2146)

### CODE REVIEW

File: acquire/acquire.py [hunk 1]
```diff
@@ -1453,6 +1453,16 @@ class Home(Module):
         ("dir", ".config", from_user_home),
         ("glob", "*/.*[akz]sh*", from_user_home),
         ("glob", "*/.config", from_user_home),
+        ("glob", ".*history", from_user_home),
+        ("glob", "*/.*history", from_user_home),
+        ("glob", ".*rc", from_user_home),
+        ("glob", ".*_logout", from_user_home),
+        ("glob", "*/.wget-hsts", from_user_home),
+        ("glob", "*/.gitconfig", from_user_home),
```
Comment (Schamper) [inline acquire/acquire.py:1461]: This file can also be in the base of a user home right?

File: acquire/acquire.py [hunk 2]
```diff
@@ -1453,6 +1453,16 @@ class Home(Module):
         ("dir", ".config", from_user_home),
         ("glob", "*/.*[akz]sh*", from_user_home),
         ("glob", "*/.config", from_user_home),
+        ("glob", ".*history", from_user_home),
+        ("glob", "*/.*history", from_user_home),
+        ("glob", ".*rc", from_user_home),
+        ("glob", ".*_logout", from_user_home),
```
Comment (Schamper) [inline acquire/acquire.py:1459]: Maybe add a small comment above `.*[akz]sh*` that this should collect most shell configuration files, and that with these you try to collect some that aren't collected with that crazy glob?

**Review by Schamper — CHANGES_REQUESTED** [review]

File: acquire/acquire.py [hunk 3]
```diff
@@ -1449,15 +1449,19 @@ def private_key_filter(path: fsutil.TargetPath) -> bool:
 @register_module("--home")
 class Home(Module):
     SPEC = [
+        # Catches most shell related configuration files
         ("glob", ".*[akz]sh*", from_user_home),
-        ("dir", ".config", from_user_home),
         ("glob", "*/.*[akz]sh*", from_user_home),
-        ("glob", "*/.config", from_user_home),
+        # Added to catch any shell related configuration file not caught with the above regex
```
Comment (Schamper) [inline acquire/acquire.py:1455]:
```suggestion
        # Added to catch any shell related configuration file not caught with the above glob
```

File: acquire/acquire.py [hunk 4]
```diff
@@ -1449,15 +1449,19 @@ def private_key_filter(path: fsutil.TargetPath) -> bool:
...
         ("glob", ".*history", from_user_home),
         ("glob", "*/.*history", from_user_home),
         ("glob", ".*rc", from_user_home),
         ("glob", ".*_logout", from_user_home),
+        # Miscellaneous configuration files
+        ("dir", ".config", from_user_home),
+        ("glob", "*/.config", from_user_home),
         ("glob", "*/.wget-hsts", from_user_home),
+        ("glob", "*.gitconfig", from_user_home),
```
Comment (Schamper) [inline acquire/acquire.py:1464]:
```suggestion
        ("glob", ".gitconfig", from_user_home),
```
I think?

**Review by Schamper — CHANGES_REQUESTED** [review]

File: acquire/acquire.py (hunk 4 above)
Comment (Poeloe) [reply acquire/acquire.py:1464]: No, it should be "file" instead of "glob", my bad

File: acquire/acquire.py [hunk 5]
```diff
@@ -1449,10 +1449,24 @@ def private_key_filter(path: fsutil.TargetPath) -> bool:
...
     SPEC = [
+        # Catches most shell related configuration files
         ("glob", ".*[akz]sh*", from_user_home),
-        ("dir", ".config", from_user_home),
         ("glob", "*/.*[akz]sh*", from_user_home),
+        # Added to catch any shell related configuration file not caught with the above glob
+        ("glob", ".*history", from_user_home),
+        ("glob", "*/.*history", from_user_home),
+        ("glob", ".*rc", from_user_home),
+        ("glob", ".*_logout", from_user_home),
```
Comment (Schamper) [inline acquire/acquire.py:1459]:
```suggestion
        ("glob", ".*rc", from_user_home),
        ("glob", "*/.*rc", from_user_home),
        ("glob", ".*_logout", from_user_home),
        ("glob", "*/.*_logout", from_user_home),
```

File: acquire/acquire.py [hunk 6]
```diff
@@ -1449,10 +1449,24 @@ def private_key_filter(path: fsutil.TargetPath) -> bool:
...
+        # Miscellaneous configuration files
+        ("dir", ".config", from_user_home),
         ("glob", "*/.config", from_user_home),
+        ("glob", "*/.wget-hsts", from_user_home),
```
Comment (Schamper) [inline acquire/acquire.py:1463]:
```suggestion
        ("glob", ".wget-hsts", from_user_home),
        ("glob", "*/.wget-hsts", from_user_home),
```

File: acquire/acquire.py [hunk 7]
```diff
@@ -1449,10 +1449,24 @@ def private_key_filter(path: fsutil.TargetPath) -> bool:
...
+        ("file", ".gitconfig", from_user_home),
+        ("glob", "*/.gitconfig", from_user_home),
+        ("file", ".selected_editor", from_user_home),
+        ("file", ".viminfo", from_user_home),
+        ("file", ".lesshist", from_user_home),
+        ("file", ".profile", from_user_home),
```
Comment (Schamper) [inline acquire/acquire.py:1469]:
```suggestion
        ("file", ".selected_editor", from_user_home),
        ("file", "*/.selected_editor", from_user_home),
        ("file", ".viminfo", from_user_home),
        ("file", "*/.viminfo", from_user_home),
        ("file", ".lesshist", from_user_home),
        ("file", "*/.lesshist", from_user_home),
        ("file", ".profile", from_user_home),
        ("file", "*/.profile", from_user_home),
```

**Review by Schamper — CHANGES_REQUESTED** [review]
I don't remember exactly, but I think we have to duplicate every entry here in case there's a configuration such as `/home/domain/username/.dotfile`, where the user directory is one folder deeper.

Ideally we solve this properly some day but for now I think we have to keep duplicating those entries.

File: acquire/acquire.py (hunk 7 above)
Comment (Schamper) [reply acquire/acquire.py:1469]: Fixed the suggestion types. I don't have a strong preference for either. Adding code to `get_spec_additions` feels like you might as well try to fix the actual underlying issue. This is hackier but works around the issue minimally.

**Review by Schamper — APPROVED** [review]

**Pull request merged**